### PR TITLE
Overwritable route file

### PR DESCRIPTION
**Pull request by tabacitu**
Allows us to easily overwrite the routes file, by placing a file in ```routes/backpack/settings.php```.

### CODE REVIEW

File: src/SettingsServiceProvider.php [hunk 1]
```diff
@@ -19,6 +19,13 @@ class SettingsServiceProvider extends ServiceProvider
     protected $defer = false;
 
     /**
+     * Where the route file lives, both inside the package and in the app (if overwritten).
+     *
+     * @var bool
```
Comment (lloy0076) [inline src/SettingsServiceProvider.php]: Is that a bool?

File: src/routes/backpack/settings.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+|--------------------------------------------------------------------------
+| Backpack\Settings Routes
+|--------------------------------------------------------------------------
+|
+| This file is where you may define or change all of the routes that are
```
Comment (lloy0076) [inline src/routes/backpack/settings.php]: We're not really changing anything here (it's defining by definition :)).

File: src/routes/backpack/settings.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+|
+*/
+
+Route::group([
+            'namespace'  => 'Backpack\Settings\app\Http\Controllers',
+            'prefix'     => config('backpack.base.route_prefix', 'admin'),
+            'middleware' => ['web', 'admin'],
+    ], function () {
+
+    // Settings
```
Comment (lloy0076) [inline src/routes/backpack/settings.php]: Is this comment necessary?

**Comment by tabacitu** [comment]
Done, merged! Thanks a lot @lloy0076 for reviewing my PRs, I think it makes for better code for all of us, in the end.

Cheers!

**Pull request merged**